Furball.Engine/Engine/Graphics/Drawables/UiElements/UiButtonDrawable.cs

```diff
@@ -0,0 +1,52 @@
+using Furball.Engine.Engine.Graphics.Drawables.Managers;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using MonoGame.Extended;
+
+namespace Furball.Engine.Engine.Graphics.Drawables.UiElements {
+    public class UiButtonDrawable : ManagedDrawable {
+        private string _text;
+        public string Text => this._text;
+        public TextDrawable TextDrawable;
+
+        private float _margin;
+        public float Margin => this._margin;
+
+        public Color OutlineColor = Color.Black;
+
+        public UiButtonDrawable(string text, byte[] font, float size, Color textColor, Color outlineColor, float margin = 5f) {
+            this.TextDrawable = new TextDrawable(font, text, size);
+
+            this.OnMove += this.RecalculateSize;
+
+            this._margin = margin;
+            this._text   = text;
+
+            this.Size = new Vector2();
+
+            this.RecalculateSize(this, this.Position);
+
+            this.TextDrawable.ColorOverride = textColor;
+            this.OutlineColor               = outlineColor;
+        }
+
+        private void RecalculateSize(object sender, Vector2 newPos) {
+            (float textDrawableSizeX, float textDrawableSizeY) = this.TextDrawable.Size;
+
+            this.Size = new Vector2(textDrawableSizeX + this._margin * 2f, textDrawableSizeY + this._margin * 2f);
+        }
+
+        public override void Draw(GameTime time, SpriteBatch batch, DrawableManagerArgs args) {
+            batch.FillRectangle(args.Position, this.Size, args.Color);
+            batch.DrawRectangle(args.Position, this.Size, this.OutlineColor, 1f, args.LayerDepth);
+
+            // FIXME: this is a bit of a hack, it should definitely be done differently
+            DrawableManagerArgs tempArgs = args;
+            tempArgs.Position.X += this._margin * args.Scale.X;
+            tempArgs.Position.Y += this._margin * args.Scale.Y;
+            tempArgs.Color      =  this.TextDrawable.ColorOverride;
+            tempArgs.Scale      /= (float)FurballGame.Instance.GraphicsDevice.Viewport.Height / FurballGame.DEFAULT_WINDOW_HEIGHT;
```

Comment:
We shouldnt be doing the calcuation every time here, we should have some sort of variable in FurballGame called WindowVerticalRatio or smth, and have that change everytime screen res gets changed

Reply:
Dividing 2 floats is only a few CPU cycles to calculate, so it isnt much of an issue performance wise, altho having the same code copied a million times is ugly, so i added a getter in FurballGame that pulls this off
Resolved in 1a3699c823ac4c6dba6c81ed07448534e99073c2

Reply:
thats kinda what i meant, sure its not too computationally intensive but its just plain ugly and repetetive code